### PR TITLE
Fix compiler warnings

### PR DESCRIPTION
**Pull request by gaborigloi**
Signed-off-by: Gabor Igloi <gabor.igloi@citrix.com>

### CODE REVIEW

File: src/main.ml [hunk 1]
```diff
@@ -120,7 +120,7 @@ let main port certfile ciphersuites =
          Lwt_log.notice "Setting up server socket" >>= fun () ->
          Lwt_unix.setsockopt sock Lwt_unix.SO_REUSEADDR true;
          let sockaddr = Lwt_unix.ADDR_INET(Unix.inet_addr_any, port) in
-         Lwt_unix.bind sock sockaddr;
+         Lwt_unix.bind sock sockaddr >>= fun () ->
```
Comment (gaborigloi) [inline src/main.ml]: This one is important IMO

File: lib/vbd_store.ml [hunk 2]
```diff
@@ -21,7 +21,7 @@ module Make(Config : sig
     Lwt.catch
       (fun () -> Lwt_unix.mkdir vbd_list_dir 0o755)
       (function
-        | Unix.(Unix_error (EEXIST, "mkdir", dir)) when dir = vbd_list_dir -> Lwt.return_unit
```
Comment (jonludlam) [inline lib/vbd_store.ml:24]: Is it just a compiler bug that it's moaning about this open?

Comment (gaborigloi) [reply lib/vbd_store.ml:24]: Maybe, I found this very puzzling. Maybe I should undo this change.

Comment (gaborigloi) [reply lib/vbd_store.ml:24]: Because the `EEXIST` variant is actually in the `Unix` module.

Comment (jonludlam) [reply lib/vbd_store.ml:24]: This is the bug that's affecting xcp-idl too - I've had to do precisely this change there. Quite annoying actually.

Comment (mseri) [reply lib/vbd_store.ml:24]: Don't despair... we will be able to move to `4.06.?` soon ;)

Comment (gaborigloi) [reply lib/vbd_store.ml:24]: So that means this PR is good to go then? :)

Comment (mseri) [reply lib/vbd_store.ml:24]: I believe so

**Comment by gaborigloi** [comment]
The Travis failure is due to `xenctrl` :( https://github.com/xapi-project/ocaml-xen-lowlevel-libs/issues/106
I'll switch it to CentOS

**Comment by mseri** [comment]
I don't understand why now it does not find `xen-api-client-lwt`, it is there: https://github.com/xapi-project/xs-opam/tree/master/packages/xs-extra/xen-api-client-lwt.master

It must be something to do with other packages

**Comment by gaborigloi** [comment]
I'm looking into it, it's an issue with the docker script:
```
Step 3/11 : RUN bash -eux -c 'git remote set-url origin git://github.com/xapi-project/xs-opam && git fetch origin master && git reset --hard origin/master'
 ---> Running in b598d9e7436b

+ git remote set-url origin git://github.com/xapi-project/xs-opam
+ git fetch origin master



warning: no common commits
From git://github.com/xapi-project/xs-opam
 * branch            master     -> FETCH_HEAD
+ git reset --hard origin/master
Checking out files: 100% (25038/25038), done.
```

**Comment by gaborigloi** [comment]
Opened https://github.com/ocaml/ocaml-ci-scripts/pull/222 t ofix the Travis issue.

**Pull request merged**